qiskit/circuit/library/n_local/qaoa_ansatz.py

```diff
@@ -75,21 +75,24 @@ def _check_configuration(self, raise_on_failure: bool = True) -> bool:
         if self._reps is None or self._reps < 1:
             valid = False
             if raise_on_failure:
-                raise AttributeError("The integer parameter p, which determines the depth "
-                                     "of the circuit, either not set or set to non-positive value")
+                raise AttributeError("The integer parameter reps, which determines the depth "
+                                     "of the circuit, needs to be >= 1 but has value {}"
+                                     .format(self._reps))
 
         num_qubits = self._cost_operator.num_qubits
         if self._initial_state is not None and self._initial_state.num_qubits != num_qubits:
             valid = False
             if raise_on_failure:
-                raise AttributeError("The number of qubits of the initial state does not match "
-                                     "the number of qubits of the cost operator")
+                raise AttributeError("The number of qubits of the initial state {} does not match "
```

Comment:
This is just a comment - not asking for any change. Since we dropped support for Python 3.5 a while back, and 3.6 is min version supported, you are free to use f-String formatting (as I used when making the comment about these messages) so that you don't need to do the .format( explicitly.